### PR TITLE
Fixes and updates to docs

### PR DESCRIPTION
**Pull request by drashna**
Fixes a number of issues (such as links and formatting) and adds some updates/clarifications for other docs.

### CODE REVIEW

File: docs/config_options.md [hunk 1]
```diff
@@ -34,6 +34,8 @@ This is a C header file that is one of the first things included, and will persi
 ## `config.h` Options
 
 ### Hardware Options
+====================
```
Comment (skullydazed) [inline docs/config_options.md]: These are conflicting styles. The `###` indicates a level 3 header, while the `====` underneath indicates a level 1 header.

File: docs/feature_advanced_keycodes.md [hunk 2]
```diff
@@ -2,7 +2,7 @@
 
 Your keymap can include keycodes that are more advanced than normal, for example shifted keys. This page documents the functions that are available to you.
 
-### Assigning Custom Names
+## Assigning Custom Names
```
Comment (skullydazed) [inline docs/feature_advanced_keycodes.md]: These level 3 headers were deliberate so they wouldn't make the TOC too confusing. With level 2 headers the TOC became big and hard to follow.

File: docs/config_options.md (hunk 1 above)
Comment (drashna) [reply docs/config_options.md]: Not a problem. Will change.

Just really want a good way to make sure these headers are LINKABLE.  And easily so.  
Because apparently, no link comes up for these... and it would make things SOO much easier.

File: docs/feature_advanced_keycodes.md (hunk 2 above)
Comment (drashna) [reply docs/feature_advanced_keycodes.md]: I figured it was something like that.... 
And H3 ... get no links (nor love, apparently)

**Comment by drashna** [comment]
Is there some way to get the H3 stuff to be linkable?

Some pages are "okay", but docs like the Config options is ... absolutely atrocious for references:
https://docs.qmk.fm/config_options.html

**Comment by skullydazed** [comment]
For the config options case I think we can remove the `config.h Options` and `rules.mk Options` options and move those H3's up to H2. I think at the time I thought there would be more structure needed there, but looking at this today I think I agree with you that those headings should be in the TOC.

**Comment by drashna** [comment]
Done.  And I think that should help with ... well helping people. 

And if we could move over to the beta version ....  that would fix many of the issues with linkability.

**Comment by skullydazed** [comment]
Nice, thanks!
